### PR TITLE
Added a new Java course

### PR DESCRIPTION
**Pull request by Nandan-18**
## What does this PR do?
Add resource(s) | Add info | Improve repo

### Description

A complete and free Java MOOC by the University of Helsinki.

### Why is this valuable (or not)?

The course covers almost all topics needed to become proficient in the language. It also follows a hands-on approach, making it easier to grasp concepts.

### How do we know it's really free?

It is a free and open-source MOOC offered by the University of Helsinki, no strings attached.

### For book lists, is it a book? For course lists, is it a course? etc.

It is a course.

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/master/CONTRIBUTING.md)
- [x] Search for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction)

## Followup

- Check the output of Travis-CI for linter errors!

### CODE REVIEW

**Review by davorpa — CHANGES_REQUESTED** [review]
When submit different works try to use same base from master instead of rebranch latest.

File: courses/free-courses-en.md [hunk 1]
```diff
@@ -372,6 +372,7 @@
 * [Java for Mobile Devices - Introducing Codename One](https://codenameone.teachable.com/p/java-for-mobile-devices-introducing-codename-one)
 * [Java Online Training \| Edureka](https://www.youtube.com/watch?v=hBh_CC5y8-s) - (Edureka)
 * [Java Programming](https://testautomationu.applitools.com/java-programming-course/) - Angie Jones (Applitools)
+* [Java Programming](https://java-programming.mooc.fi) - (University of Helsinki)
```
Comment (davorpa) [inline courses/free-courses-en.md:375]: Minor change to follow creator format guidelines.

```suggestion
* [Java Programming](https://java-programming.mooc.fi) - University of Helsinki
```

**Review by davorpa — CHANGES_REQUESTED** [review]

File: books/free-programming-books.md [hunk 2]
```diff
@@ -735,6 +735,7 @@
 * [freeCodeCamp](https://www.freecodecamp.org)
 * [FutureLearn](https://www.futurelearn.com)
 * [MIT OCW](http://ocw.mit.edu)
+* [MOOC.fi](https://www.mooc.fi/en/)
```
[the remark this answers is not in the document]
Comment (davorpa) [reply books/free-programming-books.md]: Solved by https://github.com/EbookFoundation/free-programming-books/pull/5568/commits/9eef555dfb15248f3e9ef5df7eba34c617f58474

**Review by davorpa — APPROVED** [review]

**Comment by Nandan-18** [comment]
Hey, is there anything that i have to do now?

**Comment by davorpa** [comment]
> Hey, is there anything that i have to do now?

Not necesary. It's all already solved 😉 

Ready to merge

**Pull request merged**